packages/dashboards/addon/routes/dashboards/dashboard.js

```diff
@@ -107,57 +91,55 @@ export default Route.extend({
      */
     didUpdateLayout(event, widgets) {
       if (widgets && get(widgets, 'length')) {
-        this._updateLayout(widgets);
-        let dashboard = get(this, 'currentDashboard');
-        if (get(dashboard, 'canUserEdit')) {
-          this.send('saveDashboard');
-        }
+        this.set('_stagedLayout', widgets);
       }
     },
 
+    /**
+     * Take new layout staged by didChange and set
+     * it on the dashboard model.
+     *
+     * @action commitStagedLayout
+     */
+    commitStagedLayout() {
+      this._updateLayout(get(this, '_stagedLayout'));
+    },
+
     /**
      * @action saveDashboard - saves dashboard updates
      */
     saveDashboard() {
-      run.debounce(this, this._saveDashboardFn, 100);
+      const dashboard = get(this, 'currentDashboard');
+      const widgets = get(this, 'currentDashboard.widgets');
+
+      return dashboard
+        .save()
+        .then(all(widgets.map(async widget => widget.hasDirtyAttributes && widget.save())))
```

Comment:
What happens if a widget fails to save?

Reply:
The failure will propagate out to the catch at the end of the chain and give an error message.

If a widget is delete from a dashboard and the dashboard saves, but the widget fails to delete then the dashboard will own a widget that is not referenced in its layout.  The alternative is to have a layout that references a deleted widget.  This seems like a better alternative.